### PR TITLE
Initial package, version 38.00.00.00

### PR DESCRIPTION
**Pull request by gduperrey**

### CODE REVIEW

File: SPECS/avago-mpt3sas-alt.spec [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+%define vendor_label avago
+%define driver_name mpt3sas
+
+# XCP-ng: install to the override directory
+%define module_dir override
+
+%if %undefined module_dir
+%define module_dir updates
+%endif
+
```
Comment (stormi) [inline SPECS/avago-mpt3sas-alt.spec]: Unnecessary: we define it just above.

```suggestion
```

File: SPECS/avago-mpt3sas-alt.spec [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+%prep
+%autosetup -p1 -n %{vendor_label}-%{driver_name}-%{version}
+
+%build
+%{make_build} -C /lib/modules/%{kernel_version}/build M=$(pwd) KSRC=/lib/modules/%{kernel_version}/build modules
+
+%install
+%{__install} -d %{buildroot}%{_sysconfdir}/modprobe.d
+echo 'options mpt3sas prot_mask=0x07' > %{driver_name}.conf
+%{__install} %{driver_name}.conf %{buildroot}%{_sysconfdir}/modprobe.d
```
Comment (stormi) [inline SPECS/avago-mpt3sas-alt.spec]: Doesn't this overlap with the same file provided by the avago-mpt3sas RPM?

File: SPECS/avago-mpt3sas-alt.spec [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+%define module_dir override
+
+%if %undefined module_dir
+%define module_dir updates
+%endif
+
+Summary: %{vendor_name} %{driver_name} device drivers
+Name: %{vendor_label}-%{driver_name}-alt
+Version: 38.00.00.00
+Release: 1.1%{?dist}
```
Comment (stormi) [inline SPECS/avago-mpt3sas-alt.spec:11]:
```suggestion
Release: 1%{?dist}
```

File: SPECS/avago-mpt3sas-alt.spec [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+%posttrans
+%{regenerate_initrd_posttrans}
+
+%files
+%config(noreplace) %{_sysconfdir}/modprobe.d/*.conf
+/lib/modules/%{kernel_version}/*/*.ko
+
+%changelog
+* Wed Sep 20 2023 Gael Duperrey <gduperrey@vates.tech> - 38.00.00.00-1.1
+- initial package, version 38.00.00.00-1.1
```
Comment (stormi) [inline SPECS/avago-mpt3sas-alt.spec:61]:
```suggestion
* Wed Sep 20 2023 Gael Duperrey <gduperrey@vates.tech> - 38.00.00.00-1
- initial package, version 38.00.00.00-1
```

**Review by stormi — CHANGES_REQUESTED** [review]

File: SPECS/avago-mpt3sas-alt.spec [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+version %{kernel_version}.
+
+%prep
+%autosetup -p1 -n %{vendor_label}-%{driver_name}-%{version}
+
+%build
+%{make_build} -C /lib/modules/%{kernel_version}/build M=$(pwd) KSRC=/lib/modules/%{kernel_version}/build modules
+
+%install
+# As the original driver avago-mpt3sas already exists, we don't try to replace it. To monitor if the values differs in the futur.
```
Comment (stormi) [inline SPECS/avago-mpt3sas-alt.spec]:
```suggestion
# The installed by default avago-mpt3sas RPM already provides the configuration file, so we don't include it here.
# To be re-evaluated if the values differ in the future.
```

File: SPECS/avago-mpt3sas-alt.spec [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+# %{__install} -d %{buildroot}%{_sysconfdir}/modprobe.d
+# echo 'options mpt3sas prot_mask=0x07' > %{driver_name}.conf
+# %{__install} %{driver_name}.conf %{buildroot}%{_sysconfdir}/modprobe.d
```
Comment (stormi) [inline SPECS/avago-mpt3sas-alt.spec]:
```suggestion
#%{__install} -d %{buildroot}%{_sysconfdir}/modprobe.d
#echo 'options mpt3sas prot_mask=0x07' > %{driver_name}.conf
#%{__install} %{driver_name}.conf %{buildroot}%{_sysconfdir}/modprobe.d
```

File: SPECS/avago-mpt3sas-alt.spec [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+%postun
+/sbin/depmod %{kernel_version}
+%{regenerate_initrd_postun}
+
+%posttrans
+%{regenerate_initrd_posttrans}
+
+%files
+# %config(noreplace) %{_sysconfdir}/modprobe.d/*.conf
```
Comment (stormi) [inline SPECS/avago-mpt3sas-alt.spec]:
```suggestion
#%config(noreplace) %{_sysconfdir}/modprobe.d/*.conf
```

**Review by stormi — APPROVED** [review]

**Comment by gduperrey** [comment]
Koji build (v8.2-incoming): https://koji.xcp-ng.org/taskinfo?taskID=37680

**Pull request merged**